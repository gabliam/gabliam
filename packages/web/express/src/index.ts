import { ExpressPlugin } from './express-plugin';

export * from 'http-status-codes';
export * from './express';
export * from './express-converter';
export * from './express-interceptor';
export * from './interfaces';
export * from './request';
export * from './response';
export { ExpressPlugin as default };
