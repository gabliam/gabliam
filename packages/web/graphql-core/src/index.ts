export { TYPE } from './constants';
export * from './graphql-core-plugin';
export * from './interfaces';
export * from './resolver-decorator';
export {
  Arg,
  ArgOptions,
  Args,
  ArgsType,
  Authorized,
  createParamDecorator,
  createMethodDecorator,
  Ctx,
  Directive,
  Extensions,
  registerEnumType,
  Field,
  FieldOptions,
  FieldResolver,
  Info,
  InputType,
  InputTypeOptions,
  InterfaceType,
  InterfaceTypeOptions,
  Mutation,
  ObjectType,
  ObjectTypeOptions,
  PubSub,
  Query,
  Resolver,
  Root,
  Subscription,
  SubscriptionOptions,
  createUnionType,
  UseMiddleware,
  ID, Int, ClassType, Publisher
} from 'type-graphql';
