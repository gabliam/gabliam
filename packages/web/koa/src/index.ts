import { KoaPlugin } from './koa-plugin';

export * from 'http-status-codes';
export * from './interfaces';
export * from './koa';
export * from './koa-converter';
export * from './koa-interceptor';
export * from './request';
export * from './response';
export { KoaPlugin as default };
