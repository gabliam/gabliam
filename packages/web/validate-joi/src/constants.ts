export const METADATA_KEY = {
  validate: '_validate',
};

export const ERRORS_MSGS = {
  DUPLICATED_VALIDATE_DECORATOR: `Cannot apply @Validate decorator multiple times.`,
};
