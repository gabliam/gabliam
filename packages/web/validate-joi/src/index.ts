export * from './metadatas';
export * from './validate';
export * from './constants';
export * from './validate-config';
export { ValidatePlugin as default } from './validate-plugin';
