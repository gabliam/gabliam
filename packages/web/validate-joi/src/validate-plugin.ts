import { Plugin, Scan } from '@gabliam/core';

@Plugin()
@Scan()
export class ValidatePlugin {}
