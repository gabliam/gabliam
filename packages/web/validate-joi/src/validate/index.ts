export * from './validate-interceptor';
export * from './validate-request';
