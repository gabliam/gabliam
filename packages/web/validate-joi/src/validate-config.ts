import { Config, Init } from '@gabliam/core';
import { WebConfiguration } from '@gabliam/web-core';
import { ValidateSendErrorInterceptor, ValidateInterceptor } from './validate';

@Config()
export class ValidateConfig {
  constructor(private webConfiguration: WebConfiguration) {}

  @Init()
  init() {
    this.webConfiguration.useGlobalInterceptor(
      ValidateSendErrorInterceptor,
      ValidateInterceptor
    );
  }
}
