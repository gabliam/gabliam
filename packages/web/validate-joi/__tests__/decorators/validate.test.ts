import { reflection } from '@gabliam/core';
import { Controller, Get } from '@gabliam/web-core';
import { Validate } from '../../src';

describe('Validate decorator', () => {
  test('should add Validate metadata to a method when decorated with @Validate', () => {
    @Controller('/test')
    class TestController {
      @Get(`/`)
      @Validate({
        body: [5],
      })
      test() {
        return '';
      }
    }

    expect(reflection.propMetadata(TestController)).toMatchSnapshot();
  });
  test('should add Validate metadata to a method when decorated with @Validate(validatorOptions)', () => {
    @Controller('/test')
    class TestController {
      @Get(`/`)
      @Validate({
        validator: {
          body: [5],
        },
      })
      test() {
        return;
      }
    }

    expect(reflection.propMetadata(TestController)).toMatchSnapshot();
  });

  test('should fail when decorated multiple times the same method with @Validate', () => {
    expect(function() {
      @Controller('/test')
      class TestController {
        @Get(`/`)
        @Validate({
          body: [5],
        })
        @Validate({
          body: [5],
        })
        test() {
          return;
        }
      }
      new TestController();
    }).toThrowError();
  });
});
