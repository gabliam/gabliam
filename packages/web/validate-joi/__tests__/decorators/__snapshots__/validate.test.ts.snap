// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate decorator should add Validate metadata to a method when decorated with @Validate 1`] = `
{
  "test": [
    PropDecoratorFactory {
      "method": "get",
      "path": "/",
    },
    DecoratorFactory {
      "ids": [
        [Function],
      ],
    },
    PropDecoratorFactory {
      "rules": Map {
        "body" => {
          "$_root": {
            "ValidationError": [Function],
            "_types": Set {
              "alternatives",
              "any",
              "array",
              "boolean",
              "date",
              "function",
              "link",
              "number",
              "object",
              "string",
              "symbol",
              "binary",
            },
            "allow": [Function],
            "alt": [Function],
            "alternatives": [Function],
            "any": [Function],
            "array": [Function],
            "assert": [Function],
            "attempt": [Function],
            "binary": [Function],
            "bool": [Function],
            "boolean": [Function],
            "build": [Function],
            "cache": {
              "provision": [Function],
            },
            "checkPreferences": [Function],
            "compile": [Function],
            "custom": [Function],
            "date": [Function],
            "defaults": [Function],
            "disallow": [Function],
            "equal": [Function],
            "exist": [Function],
            "expression": [Function],
            "extend": [Function],
            "forbidden": [Function],
            "func": [Function],
            "function": [Function],
            "in": [Function],
            "invalid": [Function],
            "isError": [Function],
            "isExpression": [Function],
            "isRef": [Function],
            "isSchema": [Function],
            "link": [Function],
            "not": [Function],
            "number": [Function],
            "object": [Function],
            "only": [Function],
            "optional": [Function],
            "options": [Function],
            "override": Symbol(override),
            "preferences": [Function],
            "prefs": [Function],
            "ref": [Function],
            "required": [Function],
            "string": [Function],
            "strip": [Function],
            "symbol": [Function],
            "trace": [Function],
            "types": [Function],
            "untrace": [Function],
            "valid": [Function],
            "version": "17.7.0",
            "when": [Function],
            "x": [Function],
            Symbol(@hapi/lab/coverage/initialize): [Function],
          },
          "$_super": {},
          "$_temp": {
            "ruleset": false,
            "whens": {},
          },
          "$_terms": {
            "alterations": null,
            "examples": null,
            "externals": null,
            "metas": [],
            "notes": [],
            "shared": null,
            "tags": [],
            "whens": null,
          },
          "_cache": null,
          "_flags": {
            "only": true,
          },
          "_ids": {
            "_byId": Map {},
            "_byKey": Map {},
            "_schemaChain": false,
          },
          "_invalids": null,
          "_preferences": null,
          "_refs": {
            "refs": [],
          },
          "_rules": [],
          "_singleRules": Map {},
          "_valids": {
            "_lowercase": Map {},
            "_override": true,
            "_refs": Set {},
            "_values": Set {
              5,
            },
          },
          "type": "any",
        },
      },
      "useInterceptors": true,
      "validationOptions": {
        "escapeHtml": true,
      },
    },
  ],
}
`;

exports[`Validate decorator should add Validate metadata to a method when decorated with @Validate(validatorOptions) 1`] = `
{
  "test": [
    PropDecoratorFactory {
      "method": "get",
      "path": "/",
    },
    DecoratorFactory {
      "ids": [
        [Function],
      ],
    },
    PropDecoratorFactory {
      "rules": Map {
        "body" => {
          "$_root": {
            "ValidationError": [Function],
            "_types": Set {
              "alternatives",
              "any",
              "array",
              "boolean",
              "date",
              "function",
              "link",
              "number",
              "object",
              "string",
              "symbol",
              "binary",
            },
            "allow": [Function],
            "alt": [Function],
            "alternatives": [Function],
            "any": [Function],
            "array": [Function],
            "assert": [Function],
            "attempt": [Function],
            "binary": [Function],
            "bool": [Function],
            "boolean": [Function],
            "build": [Function],
            "cache": {
              "provision": [Function],
            },
            "checkPreferences": [Function],
            "compile": [Function],
            "custom": [Function],
            "date": [Function],
            "defaults": [Function],
            "disallow": [Function],
            "equal": [Function],
            "exist": [Function],
            "expression": [Function],
            "extend": [Function],
            "forbidden": [Function],
            "func": [Function],
            "function": [Function],
            "in": [Function],
            "invalid": [Function],
            "isError": [Function],
            "isExpression": [Function],
            "isRef": [Function],
            "isSchema": [Function],
            "link": [Function],
            "not": [Function],
            "number": [Function],
            "object": [Function],
            "only": [Function],
            "optional": [Function],
            "options": [Function],
            "override": Symbol(override),
            "preferences": [Function],
            "prefs": [Function],
            "ref": [Function],
            "required": [Function],
            "string": [Function],
            "strip": [Function],
            "symbol": [Function],
            "trace": [Function],
            "types": [Function],
            "untrace": [Function],
            "valid": [Function],
            "version": "17.7.0",
            "when": [Function],
            "x": [Function],
            Symbol(@hapi/lab/coverage/initialize): [Function],
          },
          "$_super": {},
          "$_temp": {
            "ruleset": false,
            "whens": {},
          },
          "$_terms": {
            "alterations": null,
            "examples": null,
            "externals": null,
            "metas": [],
            "notes": [],
            "shared": null,
            "tags": [],
            "whens": null,
          },
          "_cache": null,
          "_flags": {
            "only": true,
          },
          "_ids": {
            "_byId": Map {},
            "_byKey": Map {},
            "_schemaChain": false,
          },
          "_invalids": null,
          "_preferences": null,
          "_refs": {
            "refs": [],
          },
          "_rules": [],
          "_singleRules": Map {},
          "_valids": {
            "_lowercase": Map {},
            "_override": true,
            "_refs": Set {},
            "_values": Set {
              5,
            },
          },
          "type": "any",
        },
      },
      "useInterceptors": true,
      "validationOptions": {
        "escapeHtml": true,
      },
    },
  ],
}
`;
